### PR TITLE
Feature/shadow most of rest

### PR DESCRIPTION
**Pull request by n-hutton**

### CODE REVIEW

File: libs/math/src/math/linalg/blas/gemv_n_vector.cpp [hunk 1]
```diff
@@ -129,7 +129,7 @@ void Blas<S, Signature(_y <= _alpha, _A, _x, _n, _beta, _y, _m),
   {
     for (j = 0; j < int(a.width()); ++j)
     {
-      Type temp = alpha * x[jx];
+      temp = alpha * x[jx];
```
Comment (ejfitzgerald) [inline libs/math/src/math/linalg/blas/gemv_n_vector.cpp:132]: Change of behaviour?

File: libs/vm/CMakeLists.txt [hunk 2]
```diff
@@ -20,3 +20,7 @@ target_link_libraries(fetch-vm
                              fetch-math
                              fetch-core
                              fetch-ledger)
+
+if (_is_clang_compiler)
+  target_compile_options(fetch-vm PRIVATE -Wshadow)
```
Comment (ejfitzgerald) [inline libs/vm/CMakeLists.txt:25]: It is not clear if this has been enabled for all libraries. If it has is it worth simple adding the configuration at the top level?

File: libs/math/src/math/linalg/blas/gemv_n_vector.cpp (hunk 1 above)
Comment (n-hutton) [reply libs/math/src/math/linalg/blas/gemv_n_vector.cpp:132]: No, it does not look like it to me?

File: libs/vm/CMakeLists.txt (hunk 2 above)
Comment (n-hutton) [reply libs/vm/CMakeLists.txt:25]: That was my intention for the next PR - not all libraries are enabled which I would do next. Then think about GCC enabling (but probably not)

**Review by ejfitzgerald — APPROVED** [review]

**Pull request merged**